layer_utils.py

```diff
@@ -30,6 +30,52 @@ def index_points(points, idx):
     return new_points
 
 
+class PConvLinearOptFunction(torch.autograd.Function):
```

Comment:
Explaining what the cuda kernel does briefly in a couple of sentences would be good here. That way, people who don't want to dig into the kernel but want to know how this particular implementation is more optimized than the naive PointConv can go through it.